spec/book_spec.rb

```diff
@@ -0,0 +1,29 @@
+require_relative '../book'
+
+describe Book do
+  let(:book) { Book.new('Introduction to CyberSecurity', 'NSUK') }
+
+  describe '#initialize' do
+    it 'creates a new book' do
+      expect(book).to be_a(Book)
+    end
+  end
+
+  describe '#add_rental' do
+    it 'adds a rental to the book' do
+      expect(book.rentals).to eq([])
+    end
+  end
+
+  describe '#serialize' do
+    it 'serializes a book' do
+      expect(book.serialize).to eq(
+        {
+          'title' => 'Alice in Wonderland',
+          'author' => 'Lewis Carrol',
+          'id' => book.id
+        }
+      )
+    end
+  end
```

Comment:
- [ ] I will implore you double check this test case because it fails.
![image](https://github.com/katfogy/oop-school-library/assets/86200356/e23d6552-4de8-4e2c-bb99-8d6994c729f6)